django/utils/cache.py

```diff
@@ -443,3 +444,35 @@ def _to_tuple(s):
     if len(t) == 2:
         return t[0].lower(), t[1]
     return t[0].lower(), True
+
+
+def invalidate_view_cache(path=None, request=None, key_prefix=None, cache=None):
+    """
+    Delete a view cache key based on either a relative URL (``path``)
+    or a request object (``request``).
+
+    The cache key is reconstructed in two steps:
+        1. A headers cache key is built using the absolute URL,
+            key prefix, and locale code.
+        2. A response cache key is then built using the absolute URL,
+            key prefix, HTTP method, and recovered headers.
+
+    The ``key_prefix`` must match the value used in the ``cache_page``
+    decorator for the corresponding view.
+
+    Either the ``path`` or ``request`` parameter must be provided.
+    If both are given, ``path`` takes precedence.
+    """
+    if not request:
+        assert path is not None, "either `path` or `request` arguments needed"
```

Comment:
Raise a `ValueError` (?) here, and test for the exception. (Rather than an assertion, to match Django code style.)